### PR TITLE
global: useful changes for developers

### PR DESCRIPTION
**Pull request by crepererum**
Some changes for developers to simplify their life. May to go `master`, `maint-2.0` or `maint-2.1`.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -3,25 +3,30 @@ root = true
 [*]
 indent_style = space
 end_of_line = lf
-insert_final_newline = true
+insert_final_newline = false
 trim_trailing_whitespace = true
```
Comment (tiborsimko) [inline .editorconfig]: Why changing this to `false`?  IIRC we agreed to keep final newlines...

**Comment by tiborsimko** [comment]
> May to go master, maint-2.0 or maint-2.1.

Yes, maint-2.0 would seem the most appropriate.

File: .editorconfig (hunk 1 above)
Comment (crepererum) [inline .editorconfig]: Then I misunderstood this during a conversation with @jirikuncar. I'll remove that change.

File: .editorconfig [hunk 2]
```diff
@@ -3,25 +3,30 @@ root = true
...
 indent_size = 2
 
 # Matches the exact files either package.json or .travis.yml
 [{package.json,.travis.yml}]
-indent_style = space
 indent_size = 2
+
+# Dockerfile
+[Dockerfile]
+indent_size = 4
```
Comment (jirikuncar) [inline .editorconfig]: missing `\n`

**Comment by jirikuncar** [comment]
:shipit: for `maint-2.0`

**Comment by crepererum** [comment]
rebased against `maint-2.0`

**Pull request merged**